### DISCUSSION TITLE
Historical Prices Pagination

### DISCUSSION THREAD

**Discussion by fabianboesiger** [fabianboesiger/ftx#92] [General]
I'm currently unable to fetch historical prices when specifying `startTime` and `endTime`, does anyone else have this problem?

[https://ftx.com/api/markets/BTC-PERP/candles?resolution=60&startTime=1638313200&endTime=1638313800&limit=10](https://ftx.com/api/markets/BTC-PERP/candles?resolution=60&startTime=1638313200&endTime=1638313800&limit=10) for example returns wrong start times.

**Comment by dovahcrow** [comment]
@fabianboesiger the paremeter should be in snake case `https://ftx.com/api/markets/BTC-PERP/candles?resolution=60&start_time=1638313200&end_time=1638313800`

**Comment by fabianboesiger** [comment]
All our getter structs are currently serialized to camelCase format, right? For requests such as `GetHistoricalPrices`, the params should be serialized to snake_case. Would it cause problems if we simply omitted `#[serde(rename_all = "camelCase")]` for requests that perform a get request?

**Reply by dovahcrow** [reply]
It looks ok to do that for GetHistoricalPrices but I'm not sure for other requests.

**Comment by SwayStar123** [comment]
Generally its the parameters that are in snake_case, but sometimes they are accepted in camelCase. It is rather inconsistent so it will have to be done on a case by case basis for every endpoint
